### PR TITLE
Update pipeline for implementation of CO2-RT text file processing

### PR DESCRIPTION
**Pull request by bpasquer**
- process of text file in the incoming_handler
- set destination path in dest_path
- new script to create netCDF file from incoming text file
- config file for netCDF attributes

### CODE REVIEW

**Comment by bpasquer** [comment]
Quite a major update. One for you maybe @lbesnard ?

**Comment by lbesnard** [comment]
sure

File: SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
+#! /usr/bin/env python
+
+import os
+import glob
```
Comment (lbesnard) [inline SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py]: not used

File: SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+import argparse
```
Comment (lbesnard) [inline SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py]: not used

File: SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+import sys
+from datetime import datetime, timedelta
```
Comment (lbesnard) [inline SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py]: timedelta not used

File: SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+import pandas as pd
+import numpy as np
+from netCDF4 import Dataset, date2num, stringtochar
```
Comment (lbesnard) [inline SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py]: date2num not used

File: SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+#import pdb
```
Comment (lbesnard) [inline SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py]: can be removed

File: SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+    """
+    Convert date and time data object into datetime
+    Input : data Frame
+    Return:
+        - time:array of decimal time from 1950-01-01T00:00:00
+        - dtime: array of datetime object
+    """
+    epoch = datetime(1950, 1, 1)
+    time = []
+    time_long = data['PcDate'] + ' ' + data['PcTime']
```
Comment (lbesnard) [inline SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py]: dataf or data ? line 299 the reference is data

File: SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py [hunk 7]
```diff
@@ -0,0 +1,344 @@
...
+    dtime = pandas.to_datetime(time_long, dayfirst=True, utc=True)
```
Comment (lbesnard) [inline SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py]: pandas should be pd as you did at the top import pandas as pd

File: SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py [hunk 8]
```diff
@@ -0,0 +1,344 @@
...
+    """
+    Create a netcdf file
+    """
+    ncfile = Dataset(netcdf_file_path, "w", format="NETCDF4")
+    config_file = os.path.join(os.getenv('DATA_SERVICES_DIR'), 'SOOP', 'SOOP_CO2', 'global_att_soop_co2.att')
+   # check use of vessel
+    if platform_code in ships():
+        vessel_name = ships()[platform_code]
+    else:
+        sys.exit("Unknow ship code '%s' from '%s' ") % nc_file
```
Comment (lbesnard) [inline SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py]: nc_file should be netcdf_file_path

File: SOOP/SOOP_CO2/dest_path.py [hunk 9]
```diff
@@ -3,61 +3,129 @@
 
 import os
 import sys
-
+import pdb
```
Comment (lbesnard) [inline SOOP/SOOP_CO2/dest_path.py]: can be removed

File: SOOP/SOOP_CO2/dest_path.py [hunk 10]
```diff
@@ -68,46 +136,46 @@ def soop_co2_dest_path(nc_file):
             return None
```
Comment (lbesnard) [inline SOOP/SOOP_CO2/dest_path.py]: line 135 , the nc_file variable does not exist
print >>sys.stderr, "Hierarchy not created for '%s'" % nc_file, should be file I guess

File: SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py (hunk 6 above)
Comment (bpasquer) [reply SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py]: Well spotted. Actually the whole function is not use and should have been deleted.

**Comment by bpasquer** [comment]
Thanks @lbesnard. Corrections are done.

File: SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py [hunk 11]
```diff
@@ -0,0 +1,321 @@
...
+         - time:array of decimal time from 1950-01-01T00:00:00
+         - dtime: array of datetime object
+    """
+    epoch = datetime(1950, 1, 1)
+    time = []
+    time_long = dataf['PcDate'].values + ' ' + dataf['PcTime'].values
+    dtime = pd.to_datetime(time_long, dayfirst=True, utc=True)
+    for t in dtime:
+        dt = datetime.strptime(datetime.strftime(t,'%d/%m/%Y %H:%M:%S'), '%d/%m/%Y %H:%M:%S')
+        time.append((dt - epoch).total_seconds())
```
Comment (lbesnard) [inline SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py]: I think this could have been done with pandas without a loop

**Pull request merged**